sor/src/main/java/com/bazaarvoice/emodb/sor/DataStoreModule.java

```diff
@@ -257,10 +257,10 @@ protected void configure() {
     }
 
     @Provides @Singleton
-    Optional<Mutex> provideMutex(DataCenterConfiguration dataCenterConfiguration, @DataStoreZooKeeper CuratorFramework curator) {
+    Optional<TableMutexManager> provideTableMutexManager(DataCenterConfiguration dataCenterConfiguration, @DataStoreZooKeeper CuratorFramework curator) {
         // We only use ZooKeeper if this is the data center that is allowed to edit table metadata (create/drop table)
         if (dataCenterConfiguration.isSystemDataCenter()) {
-            return Optional.<Mutex>of(new CuratorMutex(curator, "/lock/tables"));
+            return Optional.of(new TableMutexManager(curator, "/lock/tables"));
```

Comment:
This isn't critical, but maybe consider changing the lock path, just so that it doesn't conflict with any locks held during the transition period during a no-downtime upgrade, maybe something like "/lock/tables256" or "/lock/table-partitions" or something much better.